healthcheck/healthcheck_test.go

```diff
@@ -524,3 +524,27 @@ func TestCheckTimeout(t *testing.T) {
 		t.Errorf("Expected state change notification not received")
 	}
 }
+
+func TestCheckDryrun(t *testing.T) {
+	notify := make(chan *Notification, 10)
+	hc := NewCheck(notify)
+	hc.Blocking(true)
+	hc.Dryrun(true)
+	go hc.Run(nil)
+
+	config := NewConfig(1, &fakeChecker{succeed: false})
+	config.Interval = 250 * time.Millisecond
+	hc.Update(config)
+	hc.Stop()
```

Comment:
I feel it's better to defer hc.Stop() right after hc.Run()? Here you are counting on some results before Stop.

Reply:
You're absolutely right. I copied the pattern from TestCheckTimeout, but now since the extra sleep is removed  after config update, it's better to do it this way.